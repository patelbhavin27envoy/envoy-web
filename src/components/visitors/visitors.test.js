import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from "enzyme";

import configureMockStore from "redux-mock-store";

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Visitors } from '../Visitors'

let visitors = [{
    _id: 1,
    name : 'John Doe',
    notes : 'test first',
    signOutTime : new Date('02/09/2019').toDateString()
},
    {
        _id: 2,
        name : 'Esteban Arango',
        notes : 'Frisbee and Vegan food',
        signOutTime : new Date('04/24/2019').toDateString()
    },
    {
        _id: 3,
        name : 'Ryan Labouve',
        notes : 'Everything about Oklahoma',
        signOutTime : null,
        signedOut: null,
        _id :122
    },
    {
        _id: 4,
        name : 'Esteban Arango',
        notes : 'Frisbee and Vegan food',
        signedOut: true,
        signedIn : true,
        signOutTime : new Date('04/24/2019').toDateString()
    },
    {
        _id: 5,
        name : 'Esteban Arango',
        notes : 'Frisbee and Vegan food',
        signedOut: true,
        signedIn : false,
        signOutTime : new Date('04/24/2019').toDateString()
    }];

const visitorsWithNull = [];

// For Redux
Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureMockStore();
const store = mockStore({
    venues : visitors
});


describe("Visitor Container Component :: ", () => {

    it("for multiple entries", () => {

        // Act
        let component = renderer.create(
            <table>
            <tbody>
                <Visitors visitors={visitors}/>
            </tbody>
            </table>
        );

        // Assert
        expect(component).toMatchSnapshot();

        component = mount(
            <Visitors visitors={visitors}/>
        );

        expect(component.find('.one').length).toEqual(5);
        expect(component.find('.two').length).toEqual(5);
        expect(component.find('.three').length).toEqual(5);

    });


    it("for no visitors", () => {

        // Act
        let component = renderer.create(
            <table>
            <tbody>
             <Visitors visitors={visitorsWithNull}/>
            </tbody>
            </table>
        ).toJSON();
        // snapshot
        expect(component).toMatchSnapshot();


        component = mount(
            <Visitors visitors={visitorsWithNull}/>
        );

        expect(component.find('.flex-grow').text()).toContain('No visitors in system');
    });

});

