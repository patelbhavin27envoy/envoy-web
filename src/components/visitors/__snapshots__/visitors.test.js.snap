// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visitor Container Component ::  for multiple entries 1`] = `
<table>
  <tbody>
    <div
      className="flex-grow h-screen overflow-y-scroll"
    >
      <div
        className="mx-auto"
      >
        <div
          className="mt-8"
        >
          <table
            className="w-full"
          >
            <thead>
              <tr>
                <th
                  className="text-sm font-semibold text-grey-darker p-2 bg-grey-lightest"
                >
                  Name
                </th>
                <th
                  className="text-sm font-semibold text-grey-darker p-2 bg-grey-lightest"
                >
                  Notes
                </th>
                <th
                  className="text-sm font-semibold text-grey-darker p-1 bg-grey-lightest"
                >
                  Signed out
                </th>
              </tr>
            </thead>
            <tbody
              className="align-baseline"
            >
              <tr>
                <td
                  className="p-2 border-t border-grey-light font-mono text-xs one"
                >
                  John Doe
                   
                </td>
                <td
                  className="p-2 border-t border-grey-light font-mono text-xs two"
                >
                  test first
                </td>
                <td
                  className="p-1 border-t border-grey-light font-mono text-xs three"
                >
                  <button
                    className="btn btn-smaller btn--outline"
                    onClick={[Function]}
                  >
                    Sign out
                  </button>
                </td>
              </tr>
              <tr>
                <td
                  className="p-2 border-t border-grey-light font-mono text-xs one"
                >
                  Esteban Arango
                   
                </td>
                <td
                  className="p-2 border-t border-grey-light font-mono text-xs two"
                >
                  Frisbee and Vegan food
                </td>
                <td
                  className="p-1 border-t border-grey-light font-mono text-xs three"
                >
                  <button
                    className="btn btn-smaller btn--outline"
                    onClick={[Function]}
                  >
                    Sign out
                  </button>
                </td>
              </tr>
              <tr>
                <td
                  className="p-2 border-t border-grey-light font-mono text-xs one"
                >
                  Ryan Labouve
                   
                </td>
                <td
                  className="p-2 border-t border-grey-light font-mono text-xs two"
                >
                  Everything about Oklahoma
                </td>
                <td
                  className="p-1 border-t border-grey-light font-mono text-xs three"
                >
                  <button
                    className="btn btn-smaller btn--outline"
                    onClick={[Function]}
                  >
                    Sign out
                  </button>
                </td>
              </tr>
              <tr>
                <td
                  className="p-2 border-t border-grey-light font-mono text-xs one"
                >
                  Esteban Arango
                   
                </td>
                <td
                  className="p-2 border-t border-grey-light font-mono text-xs two"
                >
                  Frisbee and Vegan food
                </td>
                <td
                  className="p-1 border-t border-grey-light font-mono text-xs three"
                >
                  Wed Apr 24 2019
                </td>
              </tr>
              <tr>
                <td
                  className="p-2 border-t border-grey-light font-mono text-xs one"
                >
                  Esteban Arango
                   
                </td>
                <td
                  className="p-2 border-t border-grey-light font-mono text-xs two"
                >
                  Frisbee and Vegan food
                </td>
                <td
                  className="p-1 border-t border-grey-light font-mono text-xs three"
                >
                  <button
                    className="btn btn-smaller btn--outline"
                    onClick={[Function]}
                  >
                    Sign out
                  </button>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </tbody>
</table>
`;

exports[`Visitor Container Component ::  for no visitors 1`] = `
<table>
  <tbody>
    <div
      className="flex-grow h-screen overflow-y-scroll error"
    >
      <div
        className="no-visitors"
      >
        No visitors in system!! 
      </div>
    </div>
  </tbody>
</table>
`;
